micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -335,6 +338,10 @@ impl NetworkSettingsStorage for NVSStorage {
         network_settings: &[NetworkSetting],
     ) -> Result<(), Self::Error> {
         let bytes: Vec<u8> = postcard::to_allocvec(network_settings)?;
+        log::info!(
```

Comment:
Write deduplication effectiveness depends on deterministic values here. Should the values in `network_settings` be sorted somehow before invoking `postcard`?

Reply:
I can have it sort it before writing. will enforce it here as opposed to needing the caller to pre-sort it.

Reply:
I don't think it should sort by priority though. I think network name would be a better sort key.